### ISSUE TITLE
Error with tempdir() destination

### ISSUE THREAD

**Issue by kaneplusplus** [ropensci/piggyback#35]
It looks like the destination of calls to `pb_download` are not being correctly conveyed through the `dest` argument.

```{r}
library(piggyback)
> pb_download(file = "keras-environmentx86_64-apple-darwin15.6.0.tar.bz2",
+             dest = tempdir(), repo = "kaneplusplus/keras-envi")
Error: [ENOENT] Failed to make directory '../../../../var/folders': no such file or directory
> tempdir()
[1] "/var/folders/xv/3p7vnkjs4rndnqzd1zlbdvv80000gn/T//RtmpUnEs1G"
>
> dir("../../../../var/folders")
[1] "xv" "zz"
```

However, when I change the destination to anther directory, like `.` it works fine.

Session information is below.

```{r}
> library(sessioninfo)
> session_info()
─ Session info ───────────────────────────────────────────────────────────────
 setting  value
 version  R version 3.6.1 (2019-07-05)
 os       macOS Catalina 10.15.2
 system   x86_64, darwin15.6.0
 ui       X11
 language (EN)
 collate  en_US.UTF-8
 ctype    en_US.UTF-8
 tz       America/New_York
 date     2019-12-31

─ Packages ───────────────────────────────────────────────────────────────────
 package     * version   date       lib source
 assertthat    0.2.1     2019-03-21 [1] CRAN (R 3.6.0)
 cli           2.0.0     2019-12-09 [1] CRAN (R 3.6.1)
 clisymbols    1.2.0     2017-05-21 [1] CRAN (R 3.6.0)
 crayon        1.3.4     2017-09-16 [1] CRAN (R 3.6.0)
 curl          4.3       2019-12-02 [1] CRAN (R 3.6.0)
 digest        0.6.23    2019-11-23 [1] CRAN (R 3.6.0)
 fansi         0.4.0     2018-10-05 [1] CRAN (R 3.6.0)
 fs            1.3.1     2019-05-06 [1] CRAN (R 3.6.0)
 gh            1.0.1     2017-07-16 [1] CRAN (R 3.6.0)
 git2r         0.26.1    2019-06-29 [1] CRAN (R 3.6.0)
 glue          1.3.1     2019-03-12 [1] CRAN (R 3.6.0)
 httr          1.4.1     2019-08-05 [1] CRAN (R 3.6.0)
 jsonlite      1.6       2018-12-07 [1] CRAN (R 3.6.0)
 lubridate     1.7.4     2018-04-11 [1] CRAN (R 3.6.0)
 magrittr      1.5       2014-11-22 [1] CRAN (R 3.6.0)
 memoise       1.1.0     2017-04-21 [1] CRAN (R 3.6.0)
 piggyback   * 0.0.10.99 2019-12-31 [1] Github (ropensci/piggyback@9429ec8)
 R6            2.4.1     2019-11-12 [1] CRAN (R 3.6.0)
 Rcpp          1.0.3     2019-11-08 [1] CRAN (R 3.6.0)
 rlang         0.4.2     2019-11-23 [1] CRAN (R 3.6.0)
 sessioninfo * 1.1.1     2018-11-05 [1] CRAN (R 3.6.0)
 stringi       1.4.3     2019-03-12 [1] CRAN (R 3.6.0)
 stringr       1.4.0     2019-02-10 [1] CRAN (R 3.6.0)
 usethis       1.5.1     2019-07-04 [1] CRAN (R 3.6.0)
 withr         2.1.2     2018-03-15 [1] CRAN (R 3.6.0)

[1] /Library/Frameworks/R.framework/Versions/3.6/Resources/library
```

**Comment by cboettig** [comment]
Thanks for the bug report!  Yeah, I'll have to dig in, but we may need to rethink the piggyback's logic here.  Piggyback's original design tries to mimic git LFS, so it expected all data files to be in a subdirectory of a project.  We later made the API a bit more flexible with generic upload / download mechanism instead of `push/pull` using `.pbattributes` file, but the directory logic might have trouble with temp dirs.  

Can you do other dirs outside of the working dir, e.g. `..` ?  Are you running this inside an RStudio project? If so, are you running it at the project root directory or a subdirectory?

**Comment by kaneplusplus** [comment]
No problem. I really like the package.

I'm not running into problems with a small sample of other directories or absolute paths. My hunch is that it's an issue with with `fs` but so far, I've had some trouble isolating. I'll see if I can do a better job.

I'm not running within an RStudio project.

In case it is helpful for thinking about the design, my use case is similar to the one described in the S3 section of the "Piggyback comparison to alternatives" where I act as the administrator handling upload, versioning, and tracking. The consumers are analysts (or students) who need to download files via a function call but may not be interested in (or familiar with) the administration part.

**Comment by cboettig** [comment]
should be resolved by #48 (v 0.1.0)
